### PR TITLE
Sync the MaxItems constraints between the Go & JSON schemas

### PR DESCRIPTION
**Pull request by ulucinar**
### Description of your changes

Depends on: https://github.com/crossplane/upjet/pull/411

When replacing the singleton lists with embedded objects in the MR APIs, we had figured out that the MaxItems constraints in the JSON resource schema were not in-sync with the constraints declared in the Go schema. For more context, please see the description of #527.

We've previously addressed this issue by manually configuring the `MaxItems` constraints for the affected resources in #527. This also implies that if the constraints are not in sync for any future resources, then we will also need to manually configure them.

With https://github.com/crossplane/upjet/pull/411, we introduce a new schema traverser, `traverser.maxItemsSync`, that syncs the `MaxItems` constraints from a source schema to a target schema.

This PR removes the manual `MaxItems` configurations and uses `traverser.maxItemsSync` to sync the `MaxItems` constraints from the Go resource schema to the JSON resource schema. This will also help us with any future resources for which the `MaxItems` constraints are not in sync. We could in theory sync the other constraints than `MaxItems` but currently it's the only known constraint with inconsistencies causing us trouble.

With a major version bump of the provider, we would also like to get rid of the JSON resource schema that we only use during the code generation, as a long term strategy.

### Note:
As we bump the upjet version with this PR to be able to use the `traverser.maxItemsSync`, we also generate the secret references for sensitive fields under the `spec.initProvider` API tree. This has been implemented in a separate initial commit before the manual `MaxItems` constraints are removed, so that it becomes easy for the reviewers to observe how the schema traverser behaves and whether it can effective replace the current manual configuration.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### How has this code been tested

Via the uptest runs of the resources tested in #527:
- ...

Also, please observe that commit [`46b1b34e5e9cdd65edb737e1978dcf519fae6bfe`](https://github.com/crossplane-contrib/provider-upjet-gcp/pull/537/commits/46b1b34e5e9cdd65edb737e1978dcf519fae6bfe) (which invokes the `syncMaxItems` schema traverser) on top of the commit [`a2af467da6eeab7d4460f7cb95443676cd7e465f`](https://github.com/crossplane-contrib/provider-upjet-gcp/pull/537/commits/a2af467da6eeab7d4460f7cb95443676cd7e465f) (which removes the `MaxItems` configuration) produces no diff in the generated code.

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by ulucinar** [comment]
/test-examples="examples/compute/v1beta2/instancefromtemplate.yaml"

**Comment by ulucinar** [comment]
/test-examples="examples/container/v1beta2/nodepool.yaml"

File: config/kms/config.go [hunk 1]
```diff
@@ -42,24 +39,4 @@ func Configure(p *config.Provider) {
 	p.AddResourceConfigurator("google_kms_secret_ciphertext", func(r *config.Resource) {
 		r.TerraformResource.Schema["plaintext"].Sensitive = false
 	})
-
-	p.AddResourceConfigurator("google_kms_crypto_key_version", func(r *config.Resource) {
```
Comment (sergenyalcin) [inline config/kms/config.go:46]: nit: We may consider removing this configurator function because it was only responsible for syncing the schemas.

**Review by sergenyalcin — APPROVED** [review]
Thanks @ulucinar LGTM. I left a nit comment that is proposing the removal of the function that is no longer functional after the change. Also, checked the diffs in the apis and package to ensure that the configuration changes do not affect the APIs. It seems that all changes are related-to generation of SecretRefs in InitProvider blocks.

**Pull request merged**